src/openfermion/hamiltonians/_general_hubbard.py

```diff
@@ -0,0 +1,289 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""This module constructs Hamiltonians for the multiband Fermi-Hubbard model.
+"""
+
+from collections import namedtuple
+
+from openfermion.ops import FermionOperator
+from openfermion.utils import SpinPairs
+
+TunnelingParameter = namedtuple('TunnelingParameter',
+                                ('edge_type', 'dofs', 'coefficient'))
+InteractionParameter = namedtuple(
+        'InteractionParameter',
+        ('edge_type', 'dofs', 'coefficient', 'spin_pairs'))
+PotentialParameter = namedtuple('PotentialParameter',
+                                ('dof', 'coefficient'))
+
+
+def number_operator(i, coefficient=1.):
+    return FermionOperator(((i, 1), (i, 0)), coefficient)
+
+
+def interaction_operator(i, j, coefficient=1.):
+    return number_operator(i, coefficient) * number_operator(j)
+
+
+def tunneling_operator(i, j, coefficient):
+    return (FermionOperator(((i, 1), (j, 0)), coefficient) + 
+            FermionOperator(((j, 1), (i, 0)), coefficient.conjugate()))
+
+
+class FermiHubbardModel:
+    r"""A general, parameterized Fermi-Hubbard model.
+
+    The general (AKA 'multiband') Fermi-Hubbard model has `k` degrees of
+    freedom per site in a lattice.
+    The lattice can have periodic boundary conditions or not.
+    For a lattice with `n` sites, there are `N = k * n` spatial orbitals.
+    In the standard Fermi-Hubbard model (which we call the "spinful" model),
+    there is room for an "up" fermion and a "down" fermion at each site on the
```

Comment:
Perhaps more accurate to say "at each spatial orbital" rather than each "site", since you say there are n site sand N =k*n spatial orbitals.